buildSrc/src/main/java/org/opensearch/gradle/PublishPlugin.java

```diff
@@ -128,6 +128,11 @@ public String call() throws Exception {
             // have to defer this until archivesBaseName is set
             project.afterEvaluate(p -> publication.setArtifactId(getArchivesBaseName(project)));
 
+            if (project.getPlugins().hasPlugin("opensearch.java")) {
+                publication.artifact(project.getTasks().getByName("sourcesJar"));
+                publication.artifact(project.getTasks().getByName("javadocJar"));
+            }
```

Comment:
Minor: Looks like `getPluginManager()` is preferred rather than `getPlugins()`, according to https://discuss.gradle.org/t/using-plugin-manager-vs-plugin-container-in-java-plugin/34682. While I didn't find the reason for that, so it's fine.